src/main.js

```diff
@@ -0,0 +1,98 @@
+// Use Node-style imports for dependencies.
+const axios = require('axios');
+const readline = require('readline');
+
+// Setup for the readline library.
+const reader = readline.createInterface({
+  input: process.stdin,
+  output: process.stdout
+});
+
+// Custom error class to represent a generic failure in the Pets API.
+class PetsApiError extends Error {
+}
+
+// Custom error class to represent failing to find a pet in the Pets API. 
+class PetNotFound extends PetsApiError {
+}
+
+// Option functions.
+const exit = () => {
+  console.log("Thank you for using the Ada Pets Adoption App!");
+  process.exit();
+}
+
+const listPets = () => {
+  // Fill in as part of Wave 1.
+  throw new PetsApiError("danger! danger!");
+}
+
+// Use a closure to make `selectedPet` private.
+const [selectPet, showDetails, removePet] = (() => {
```

Comment:
Unless we bump up the spreads/destructuring/etc material we have, students will not read this well -- any thoughts on how to make this more clear?

I'm unsure if throwing a reading will help with this, or if it just needs to be rewritten a different way

Reply:
Or honestly, a very elaborate comment may be a possibility; I think that this serves "students that just want to get the project done," and doesn't serve students that have the opinion that projects should only have material that they've been exposed to.

It may be interesting to do a long comment and a "for more information, read up on:" kind of thing?